server/src/main/java/org/tctalent/server/model/db/DuolingoCoupon.java

```diff
@@ -16,54 +16,78 @@
 
 package org.tctalent.server.model.db;
 
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
 import java.time.LocalDateTime;
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToOne;
-import javax.persistence.SequenceGenerator;
-import javax.persistence.Table;
 import lombok.Getter;
 import lombok.Setter;
 import org.springframework.lang.Nullable;
 
+/**
+ * Represents a Duolingo coupon that can be assigned to candidates for accessing Duolingo tests.
+ * These coupons are managed and tracked by the system to ensure their proper usage and expiration
+ * handling.
+ */
 @Getter
 @Setter
 @Entity
 @Table(name = "duolingo_coupon")
 @SequenceGenerator(name = "seq_gen", sequenceName = "coupon_id_seq", allocationSize = 1)
 public class DuolingoCoupon {
 
+  /**
+   * Unique identifier for the coupon.
+   */
   @Id
   @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_gen")
   @Column(name = "id")
   private Long id;
 
+  /**
+   * Unique code representing the Duolingo coupon. This code is used by candidates to redeem the
+   * coupon.
+   */
   @Column(name = "coupon_code", unique = true)
   private String couponCode;
-
+  /**
+   * The candidate to whom this coupon is assigned. Can be null if the coupon is not yet assigned.
+   */
   @Nullable
   @ManyToOne
   @JoinColumn(name = "candidate_id")
   private Candidate candidate;
-
+  /**
+   * The date and time when the coupon expires. After this date, the coupon becomes invalid.
+   */
   @Column(name = "expiration_date")
   private LocalDateTime expirationDate;
-
+  /**
+   * The date and time when the coupon was sent to the candidate. Can be null if the coupon has not
+   * been sent yet.
+   */
   @Nullable
   @Column(name = "date_sent")
   private LocalDateTime dateSent;
-
-  @Nullable
-  @Column(name = "assignee_email")
-  private String assigneeEmail;
-
+  /**
+   * Current status of the coupon (e.g., "Available", "Assigned", "Redeemed"). This field tracks the
+   * lifecycle of the coupon.
+   */
   @Column(name = "coupon_status")
-  private String couponStatus;
-
+  @Enumerated(EnumType.STRING)
+  private CouponStatus couponStatus;
+  /**
+   * The test status associated with this coupon (e.g., "Completed", "In Progress"). Can be null if
+   * no test activity has occurred.
+   */
   @Nullable
   @Column(name = "test_status")
   private String testStatus;
```

Comment:
I'm not sure if we need this property - at least not as a property of a coupon?

We have an existing candidate_exam table where we can create a Duolingo exam entry for the candidate. We can put any test-related updates there - there's an existing notes field which may suffice.

If there's no good reason why we need this property in the coupon at this stage - please can you remove it for now?